src/sgm_stereo_node.cpp

```diff
@@ -165,16 +115,47 @@ void StereoSGMNode::computeSGMStereoDisparity(const sensor_msgs::ImageConstPtr&
                                               const image_geometry::StereoCameraModel& model,
                                               stereo_msgs::DisparityImage& disp_msg)
 {
+  // Get OpenCV Image Mats (grayscale)
   const auto l_image = cv_bridge::toCvShare(l_image_msg, sensor_msgs::image_encodings::MONO8)->image;
   const auto r_image = cv_bridge::toCvShare(r_image_msg, sensor_msgs::image_encodings::MONO8)->image;
 
+  cv::Mat left_gray_img = l_image.clone();
+  cv::Mat right_gray_img = r_image.clone();
+
+  // ------------------- Compute Mask for Textureless Left regions --------------
+  cv::Mat left_grad;
+  extractSobelVerticalEdges(left_gray_img, "Left Gray ", left_grad);
+  cv::Mat left_textureless_mask  = cv::Mat::zeros(left_gray_img.size(), CV_8U);
+  extractTextureLessRegion(left_grad, left_textureless_mask);
+
+  if (display_debugging_images_)
+  {
+    cv::Mat left_gray_img_textured  = cv::Mat::zeros(left_gray_img.size(), CV_8U);
+    left_gray_img.copyTo(left_gray_img_textured, left_textureless_mask);
+
+    if (rotate_images_)
+    {
+      cv::flip(left_gray_img_textured, left_gray_img_textured, -1);
+    }
+
+    cv::imshow("left_gray_img_textured", left_gray_img_textured);
```

Comment:
Could you use a ros::Publisher instead, please? That way it's easier to debug from a robot.

Reply:
Not necessary since you can ssh with -X flag into robot and visualize as necessary.

Reply:
But this will always pop up into the window, while with a publisher its the user that connects to see the images.

What happens when you run this in a system w/o X? Spams with errors, does nothing...?

Reply:
It's only enabled when the flag is so this is optional.